tomcat/src/main/java/org/apache/coyote/http11/response/ResponseResource.java

```diff
@@ -2,70 +2,138 @@
-			String responseBody = createResponseBody("/index.html");
-			return new ResponseResource(responseBody, "/index.html");
-		}
-
-		if (path.urlPath().equals("/login")) {
-			QueryStrings queryStrings = path.queryStrings();
-			String account = queryStrings.getQueryStringValueByKey("account");
-			String password = queryStrings.getQueryStringValueByKey("password");
-			login(account, password);
-			String responseBody = createResponseBody("/login.html");
-			return new ResponseResource(responseBody, "/login.html");
-		}
-
-		String responseBody = createResponseBody(path.urlPath());
-		return new ResponseResource(responseBody, path.urlPath());
-	}
-
-	public String getResponseBody() {
-		return responseBody;
-	}
-
-	public static void login(String account, String password) {
-		final User user = InMemoryUserRepository.findByAccount(account).orElseThrow(NoSuchElementException::new);
-		if (user.checkPassword(password)) {
-			log.info("user {}", user);
-		}
-	}
-
-	public String parseExtension() {
-		return urlPath.substring(urlPath.lastIndexOf("."));
-	}
-
-	private static String createResponseBody(String urlPath) throws IOException {
-		URL resource = ResponseResource.class.getClassLoader().getResource("static" + urlPath);
-		String responseBody = new String(Files.readAllBytes(new File(resource.getFile()).toPath()));
-		return responseBody;
-	}
-
-	public String getUrlPath() {
-		return urlPath;
-	}
+    private final String responseBody;
+    private final String filePath;
+    private final StatusCode statusCode;
+    private final Cookies cookies;
+
+    private ResponseResource(final String responseBody, String filePath, StatusCode statusCode, Cookies cookies) {
+        this.responseBody = responseBody;
+        this.filePath = filePath;
+        this.statusCode = statusCode;
+        this.cookies = cookies;
+    }
+
+    public static ResponseResource of(final Path path, final RequestBodies requestBodies, final HttpMethod httpMethod, final Cookies cookies) throws IOException {
+        if (HttpMethod.POST.name().equals(httpMethod.name())) {
+            return postResponseResource(path, requestBodies, cookies);
+        }
+        return getResponseResource(path, cookies);
+    }
+
+    private static ResponseResource postResponseResource(Path path, RequestBodies requestBodies, Cookies cookies) throws IOException {
+        if (path.urlPath().equals("/register")) {
+            String account = requestBodies.getRequestBodyValueByKey("account");
+            String password = requestBodies.getRequestBodyValueByKey("password");
+            String email = requestBodies.getRequestBodyValueByKey("email");
+            InMemoryUserRepository.save(new User(account, password, email));
+            String responseBody = new ResponseBody("/index.html").getResponseBody();
+            return new ResponseResource(responseBody, "/index.html", StatusCode.FOUND, Cookies.emptyCookies());
+        }
+        if (path.urlPath().equals("/login")) {
```

Comment:
```suggestion
        if (path.urlPath().equals("/register")) {
            String account = requestBodies.getRequestBodyValueByKey("account");
            String password = requestBodies.getRequestBodyValueByKey("password");
            String email = requestBodies.getRequestBodyValueByKey("email");
            
            InMemoryUserRepository.save(new User(account, password, email));
            
            String responseBody = new ResponseBody("/index.html").getResponseBody();
            return new ResponseResource(responseBody, "/index.html", StatusCode.FOUND, Cookies.emptyCookies());
        }
        
        if (path.urlPath().equals("/login")) {
```
공백 라인을 활용하여 문맥을 나누어 보세요 😄